src/transport/smtp/client/async_net.rs

```diff
@@ -381,6 +381,50 @@ impl AsyncNetworkStream {
             InnerAsyncNetworkStream::None => false,
         }
     }
+
+    pub fn peer_certificate(&self) -> Result<Vec<u8>, Error> {
+        match &self.inner {
+            #[cfg(feature = "tokio1")]
+            InnerAsyncNetworkStream::Tokio1Tcp(_) => {
+                Err(error::client("Connection is not encrypted"))
+            }
+            #[cfg(feature = "tokio1-native-tls")]
+            InnerAsyncNetworkStream::Tokio1NativeTls(stream) => Ok(stream
+                .get_ref()
+                .peer_certificate()
+                .map_err(|e| error::tls(e))?
+                .unwrap()
```

Comment:
As far as I can understand from the docs these streams return Options because the connection may not have yet been established, but here we can assume that is true

Reply:
Sounds right to me